Starter with 5.0.0-SNAPSHOT uses incorrect launcher class in pom.xml

### Version

5.0.0-SNAPSHOT

### Context

The generated sample with 5.0.0-SNAPSHOT uses incorrect launcher class in pom.xml

### Steps to reproduce

1. Go to https://start.vertx.io/
2. Select version 5.0.0-SNAPSHOT
3. Generate the sample
4. Extract the sample
5. Build the sample - `mvn clean package`
6. Run the sample `java -jar target/sample-1.0.0-SNAPSHOT-fat.jar`

the commands ends with the following error
```
Error: Could not find or load main class io.vertx.core.Launcher
Caused by: java.lang.ClassNotFoundException: io.vertx.core.Launcher
```

The reason is because the `pom.xml` uses the following property:
```
    <launcher.class>io.vertx.core.Launcher</launcher.class>
```

When changed to the following the sample works:

```
<launcher.class>io.vertx.launcher.application.VertxApplication</launcher.class>
```

--- a/src/main/resources/templates/pom.xml.ftl
+++ b/src/main/resources/templates/pom.xml.ftl
@@ -31,7 +31,11 @@
 </#if>
 
     <main.verticle>${packageName}.MainVerticle</main.verticle>
+<#if vertxVersion?starts_with("5.")>
+    <launcher.class>io.vertx.launcher.application.VertxApplication</launcher.class>
+<#else>
     <launcher.class>io.vertx.core.Launcher</launcher.class>
+</#if>
   </properties>
 
   <dependencyManagement>
@@ -224,11 +228,9 @@
         <version>${exec-maven-plugin.version}</version>
 </#noparse>
         <configuration>
-<#if vertxVersion?starts_with("5.")>
-          <mainClass>io.vertx.launcher.application.VertxApplication</mainClass>
-<#else>
-          <mainClass>io.vertx.core.Launcher</mainClass>
-</#if>
+<#noparse>
+          <mainClass>${launcher.class}</mainClass>
+</#noparse>
           <arguments>
 <#if vertxVersion?starts_with("4.")>
             <argument>run</argument>

